sycl/source/detail/config.hpp

```diff
@@ -179,6 +179,28 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_RT_WARNING_LEVEL> {
+  using BaseT = SYCLConfigBase<SYCL_RT_WARNING_LEVEL>;
+
+public:
+  static unsigned int get() {
+    static bool Initialized = false;
+
+    static unsigned int Level = 0; // No RT warnings by default
+
+    // Configuration parameters are processed only once, like reading a string
+    // from environment and converting it into a typed object.
+    if (Initialized)
+      return Level;
+
+    const char *ValStr = BaseT::getRawValue();
+    int SignedLevel = (ValStr ? std::atoi(ValStr) : 0);
+    Level = SignedLevel >= 0 ? SignedLevel : 0;
+    Initialized = true;
+    return Level;
```

Comment:
```suggestion
    static unsigned int Level =[this]() {
      const char *ValStr = BaseT::getRawValue();
      int SignedLevel = (ValStr ? std::atoi(ValStr) : 0);
      return SignedLevel >= 0 ? SignedLevel : 0;
    }
    return Level;
```
Minor. Maybe something like this would be more pretty.

Reply:
Looks unusual and... yes, more compact. I made it working by fixing couple of syntax issues: removed "this" and added "();". Thank you.

Reply:
Please see the fix: f05954d81ae5dcb094700425238092b4111e35be